code/game/gamemodes/clock_cult/clock_ratvar.dm

```diff
@@ -0,0 +1,240 @@
+			arrival_text = "[max(300 - progress_in_seconds, 0)]s"
+		user << "<span class='big'><b>Seconds until Ratvar's arrival:</b> [arrival_text]</span>"
+		switch(progress_in_seconds)
+			if(-INFINITY to 100)
+				user << "<span class='heavy_brass'>It's still opening.</span>"
+			if(100 to 250)
+				user << "<span class='heavy_brass'>It's reached the Celestial Derelict and is drawing power from it.</span>"
+			if(250 to INFINITY)
+				user << "<span class='heavy_brass'>Ratvar is coming through the gateway!</span>"
+	else
+		switch(progress_in_seconds)
+			if(-INFINITY to 100)
+				user << "<span class='warning'>It's a swirling mass of blackness.</span>"
+			if(100 to 250)
+				user << "<span class='warning'>It seems to be leading somewhere.</span>"
+			if(250 to INFINITY)
+				user << "<span class='warning'><b>Something is coming through!</b></span>"
+
+/obj/structure/clockwork/massive/ratvar
+	name = "Ratvar, the Clockwork Justiciar"
+	desc = "<span class='userdanger'>What is what is what are what real what is all a lie all a lie it's all a lie why how can what is</span>"
+	clockwork_desc = "<span class='large_brass'><b><i>Ratvar, the Clockwork Justiciar, your master eternal.</i></b></span>"
+	icon = 'icons/effects/512x512.dmi'
+	icon_state = "ratvar"
+	pixel_x = -248
+	pixel_y = -248
+	takes_damage = FALSE
+	var/mob/living/prey //Whoever Ratvar is chasing
+	var/clashing = FALSE //If Ratvar is FUCKING FIGHTING WITH NAR-SIE
+
+/obj/structure/clockwork/massive/ratvar/New()
+	..()
+	SSobj.processing += src
+	world << "<span class='heavy_brass'><font size=15>\"I AM FREE!\"</font></span>"
+	ratvar_awakens = TRUE
+	spawn(50)
+		SSshuttle.emergency.request(null, 0.3)
+
+/obj/structure/clockwork/massive/ratvar/Destroy()
+	SSobj.processing -= src
+	world << "<span class='heavy_brass'><font size=7>\"NO! I will not... be...</font> <font size=6>banished...</font> <font size=5>again...\"</font></span>"
+	ratvar_awakens = FALSE
+	..()
+
+/obj/structure/clockwork/massive/ratvar/process()
+	for(var/atom/A in range(7, src))
+		A.ratvar_act()
+	if(clashing) //Doesn't move during a clash
+		return 0
+	var/dir_to_step_in = pick(cardinal)
+	if(!prey)
+		var/list/meals = list()
+		for(var/mob/living/L in living_mob_list)
+			if(L.z == z && !is_servant_of_ratvar(L) && L.mind)
+				meals += L
+		if(meals.len)
+			prey = pick(meals)
+			prey << "<span class='heavy_brass'><font size=5>\"You will do.\"</font></span>\n\
+			<span class='userdanger'>Something very large and very malevolent begins lumbering its way towards you...</span>"
+	else
+		if(prob(10) || prey.stat == DEAD || is_servant_of_ratvar(prey) || prey.z != z)
+			prey << "<span class='heavy_brass'><font size=5>\"How dull. Leave me.\"</font></span>\n\
+			<span class='userdanger'>You feel tremendous relief as a set of horrible eyes loses sight of you...</span>"
+			prey = null
+		else
+			if(prob(75))
+				dir_to_step_in = get_dir(src, prey)
+	for(var/i in 1 to 2)
+		loc = get_step(src, dir_to_step_in)
```

Comment:
I'd prefer just doing this once so you get the gliding effect. also use forceMove(get_step(src, dir_to_step_in))